gcp_variant_transforms/libs/bigquery_row_generator.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handles generating bigquery row from variants."""
+
+from __future__ import absolute_import
+
+import copy
+import json
+from typing import Dict, Any  # pylint: disable=unused-import
+
+from gcp_variant_transforms.beam_io import vcfio
+from gcp_variant_transforms.libs import bigquery_schema_descriptor  # pylint: disable=unused-import
+from gcp_variant_transforms.libs import bigquery_util
+from gcp_variant_transforms.libs import processed_variant  # pylint: disable=unused-import
+from gcp_variant_transforms.libs import vcf_field_conflict_resolver  # pylint: disable=unused-import
+from gcp_variant_transforms.libs import vcf_header_parser # pylint: disable=unused-import
+
+
+# Maximum size of a BigQuery row is 10MB. See
+# https://cloud.google.com/bigquery/quotas#import for details.
+# We set it to 10MB - 10KB to leave a bit of room for error in case jsonifying
+# the object is not exactly the same in different libraries.
+_MAX_BIGQUERY_ROW_SIZE_BYTES = 10 * 1024 * 1024 - 10 * 1024
+# Number of bytes to add to the object size when concatenating calls (i.e.
+# to account for ", "). We use 5 bytes to be conservative.
+_JSON_CONCATENATION_OVERHEAD_BYTES = 5
+
+
+class BigQueryRowGenerator(object):
+  """Class to generate bigquery row from a variant."""
+
+  def __init__(
+      self,
+      schema_descriptor,  # type: bigquery_schema_descriptor.SchemaDescriptor
+      conflict_resolver=None,
+      # type: vcf_field_conflict_resolver.ConflictResolver
+      ):
+    # type: (...) -> None
+    self._schema_descriptor = schema_descriptor
+    self._conflict_resolver = conflict_resolver
```

Comment:
should `allow_incompatible_records` and `omit_empty_sample_calls` also be here? that way you don't need to pass them around. It also seems like you're missing them from the args when calling `get_rows` in `variant_to_bigquery` and it's not covered by tests. Is it possible to add a test for it?

Reply:
SGTM.